src/main/java/org/wildfly/security/auth/server/ServerAuthenticationContext.java

```diff
@@ -1057,6 +1037,7 @@ boolean isStarted() {
     }
 
     static final class AuthorizedState extends State {
+        private final SecurityDomain domain;
```

Comment:
Since the domain is always on the identity, we don't strictly need an extra field just to cache it; in all but the most unusual cases, delegating through a simple getter is equally performant to having a field value.  Not super important though, just something to fix next time we're running through this code.